# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/TestingFunctions.cpp
# Commit: 32f22e0ad5c5
# Full Hash: 32f22e0ad5c578e1c363433cadb0551fcca833a3
# Author: Jason Orendorff <jorendorff@mozilla.com>
# Date: 2018-03-12 23:12:01
# Description:
#   Bug 1444339 - Crash using baselineCompile() in a fresh compartment. r=nbp.
# ==============================================================================

diff -r b3bf314ae495 -r 32f22e0ad5c5 js/src/builtin/TestingFunctions.cpp
--- a/js/src/builtin/TestingFunctions.cpp	Mon Mar 12 17:48:45 2018 +0200
+++ b/js/src/builtin/TestingFunctions.cpp	Fri Mar 09 09:33:30 2018 -0600
@@ -5113,6 +5113,8 @@
         return ReturnStringCopy(cx, args, "baseline disabled");
     if (!script->canBaselineCompile())
         return ReturnStringCopy(cx, args, "can't compile");
+    if (!cx->compartment()->ensureJitCompartmentExists(cx))
+        return false;
 
     jit::MethodStatus status = jit::BaselineCompile(cx, script, forceDebug);
     switch (status) {
